pandas/_libs/lib.pyx

```diff
@@ -1089,6 +1090,45 @@ def is_float(obj: object) -> bool:
     return util.is_float_object(obj)
 
 
+def is_hashable(obj: object) -> bool:
+    """
+    Return True if hash(obj) will succeed, False otherwise.
+
+    Some types will pass a test against collections.abc.Hashable but fail when
+    they are actually hashed with hash().
+
+    Distinguish between these and other types by trying the call to hash() and
+    seeing if they raise TypeError.
+
+    Returns
+    -------
+    bool
+
+    Examples
+    --------
+    >>> import collections
+    >>> from pandas.api.types import is_hashable
+    >>> a = ([],)
+    >>> isinstance(a, collections.abc.Hashable)
+    True
+    >>> is_hashable(a)
+    False
+    """
+    # Unfortunately, we can't use isinstance(obj, collections.abc.Hashable),
+    # which can be faster than calling hash. That is because numpy scalars
+    # fail this test.
+
+    # Reconsider this decision once this numpy bug is fixed:
+    # https://github.com/numpy/numpy/issues/5562
```

Comment:
looks like this issue has been closed.  maybe this can be reconsidered?

Reply:
I've benchmark a version using `isinstance(obj, collections.abc.Hashable)` it is considerably slower than the current python implementation